addons/mail/wizard/mail_compose_message.py

```diff
@@ -511,18 +510,31 @@ def _process_state(self, mail_values_dict):
             elif optout_emails and mail_to in optout_emails:
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_optout'
-            elif done_emails and mail_to in done_emails and not mailing_document_based:
+            elif mail_to in done_emails:
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_dup'
+            elif mail_to in sent_emails.keys():
```

Comment:
Feeling like this should be done after the check on void or falsy, as we don't want to do the comparison for those emails